packages/mds-service-helpers/client/index.ts

```diff
@@ -39,3 +40,11 @@ export const getServiceResult = async <R>(request: Promise<ServiceResponse<R>>):
   }
   return response.result
 }
+
+// eslint-reason type inference requires any
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyServiceResponseMethod = AnyFunction<Promise<ServiceResponse<any>>>
+
+export const UnwrapServiceResult = <M extends AnyServiceResponseMethod>(method: M) => async (
+  ...args: Parameters<M>
+): Promise<ReturnType<M> extends Promise<ServiceResponse<infer R>> ? R : unknown> => getServiceResult(method(...args))
```

Comment:
I've been staring at this for a while, deep for my brain.  I appreciate seeing your conversion of the jurisdiction client interface to see what it's doing.  What's your take on commenting these types of things?

Reply:
Some comments probably wouldn’t hurt.

Reply:
But basically it’s using type inference to convert a function that returns `Promise<ServiceResult<R>>` to a function with identical parameters that returns `Promise<R>`

Reply:
Thanks even that one-liner is helpful.  And points me to go learn about inference.  I'm often asking myself what kind of typing construct is this? To be able to go and learn more myself. The advanced TS typing docs feel vast still.

Reply:
@twelch I've found the typescript handbook to be fairly poorly structured. A lot of the super-cool advanced type usage (like `infer`) came around in [TS 2.8](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html) but while the additions are documented there, it's a bit tricky to find them when just scanning the docs.